ops/services/app_service/metabase/main.tf

```diff
@@ -1,3 +1,73 @@
+locals {
+  app_setting_defaults = {
+    "MB_DB_CONNECTION_URI"           = var.postgres_url
+    "WEBSITE_VNET_ROUTE_ALL"         = 1
+    "WEBSITE_DNS_SERVER"             = "168.63.129.16"
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = var.ai_instrumentation_key
+  }
+
+  grant_command = <<EOF
+  psql \
+  --host ${var.postgres_server_name} \
+  --port ${var.postgres_port} \
+  --username ${var.administrator_login} \
+  --dbname "${var.postgres_db_name}" \
+  --command "
+    CREATE ROLE ${data.azurerm_key_vault_secret.postgres_readonly_user.value} with LOGIN ENCRYPTED PASSWORD
+    '${data.azurerm_key_vault_secret.postgres_readonly_pass.value}';
+    GRANT USAGE ON SCHEMA public TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility_device_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE device_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE organization TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE patient_answers TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE patient_link TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE data_hub_upload TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE time_of_consent TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE specimen_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE device_specimen_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility_device_specimen_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE test_order TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+
+    GRANT SELECT (internal_id, created_at, updated_at, last_seen, is_deleted) ON TABLE api_user
```

Comment:
I feel like we should include the username, if not the person name—in the case, we'd be exposing the identities of patients who use patient link, but in the former I think we're just exposing our users' login e-mails and the UUID of the patients, which should be PHI-free.

Reply:
Is there a username field for api_user? I didn't see one when querying the local schema, but things might be different in the production/public schema.

Reply:
It's "login_email", which was a bad choice and I have regrets. 😞

Reply:
Ahhh I see, thank you. Added visibility for that.